test/repositories/survey_repository_test.dart

```diff
@@ -58,4 +59,48 @@ main() async {
           throwsA(isA<NotFound>()));
     });
   });
+
+  group('Validate getSurveyById', () {
+    final singleSurveyFile = File('test/test_resources/single_survey.json');
+    final singleSurveyJson = jsonDecode(singleSurveyFile.readAsStringSync());
+    final singleSurveyData = singleSurveyJson['data'];
+
+    final testedSurveyRepository = SurveyRepositoryImpl(mockGraphQLClient);
+
+    test('When getSurveyById successfully, it returns a Survey', () async {
+      when(mockGraphQLClient.query(any)).thenAnswer(
+        (_) async => QueryResult.optimistic(
+          data: singleSurveyData,
+        ),
+      );
+
+      final result = await testedSurveyRepository.getSurveyById("any");
+
+      expect(result, isA<Survey>());
+      expect(result.title, "Scarlett Bangkok");
+    });
+
+    test('When get Survey successfully, it has added extra data', () async {
+      when(mockGraphQLClient.query(any)).thenAnswer(
+        (_) async => QueryResult.optimistic(
+          data: singleSurveyData,
+        ),
+      );
+      final result = await testedSurveyRepository.getSurveyById("any");
+
+      expect(result.hdCoverImageUrl, result.coverImageUrl + "l");
+      expect(result.questions.first.hdCoverImageUrl,
+          result.questions.first.coverImageUrl + "l");
```

Comment:
can we use `hdCoverImageUrl` instead? 🤓

Reply:
hdCoverImageUrl is a computed attribute, so we are trying to test its logic against the original data 🤔

Reply:
to test its logic, better to test it on survey_test class instead?